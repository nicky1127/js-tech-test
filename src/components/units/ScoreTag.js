import React from "react";
import PropTypes from "prop-types";
import clsx from "clsx";
import { makeStyles } from "@material-ui/core/styles";
const useStyles = makeStyles({
  root: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "rgb(6,48,133)",
    color: "#fff",
    width: "70%",
    height: "30px",
    borderRadius: "5px"
  }
});

const ScoreTag = props => {
  const classes = useStyles();
  const { scores } = props;
  return (
    <div
      className={clsx("score", classes.root)}
      data-testid="scoreTag"
    >{`${scores.home}-${scores.away}`}</div>
  );
};

ScoreTag.defaultProps = { scores: {} };

ScoreTag.propTypes = {
  scores: PropTypes.shape({
    home: PropTypes.number,
    away: PropTypes.number
  })
};

export default ScoreTag;
