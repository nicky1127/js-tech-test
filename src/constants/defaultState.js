const state = {
  liveEvents: [],
  event: {},
  markets: {},
  isOddsDecimal: false,
  loading: false,
  error: {},
  selections: []
};

export default state;
