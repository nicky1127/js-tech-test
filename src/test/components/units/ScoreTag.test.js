import React from "react";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import ScoreTag from "components/units/ScoreTag";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import defaultState from "constants/defaultState";

beforeEach(() => {
  jest.clearAllMocks();
  cleanup();
});

const props = {};

const state = {
  ...defaultState
};
const mockStore = configureMockStore();
const store = mockStore(state);

describe("ScoreTag", () => {
  it("Snapshot matchs", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ScoreTag {...props} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("ScoreTag should render one [data-testid=scoreTag]", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <ScoreTag {...props} />
      </Provider>
    );
    const container = getByTestId("scoreTag");
    expect(container).toBeInTheDocument();
  });
});
