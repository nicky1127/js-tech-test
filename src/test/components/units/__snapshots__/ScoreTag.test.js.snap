// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreTag Snapshot matchs 1`] = `
<div
  className="score makeStyles-root-1"
  data-testid="scoreTag"
>
  undefined-undefined
</div>
`;
